src/pydvl/parallel/backends/joblib.py

```diff
@@ -19,50 +20,72 @@
 logger = logging.getLogger(__name__)
 
 
-class JoblibParallelBackend(BaseParallelBackend, backend_name="joblib"):
+class JoblibParallelBackend(ParallelBackend, backend_name="joblib"):
     """Class used to wrap joblib to make it transparent to algorithms.
 
-    It shouldn't be initialized directly. You should instead call
-    [init_parallel_backend()][pydvl.parallel.backend.init_parallel_backend].
-
-    ??? Example
-        ``` python
-        from pydvl.parallel import init_paralle_backend, ParallelConfig
-        config = ParallelConfig(backend="joblib")
-        parallel_backend = init_parallel_backend(config)
-        ```
-
-    ??? Example
+    !!! Example
         ``` python
-        import joblib
-        from pydvl.parallel import init_paralle_backend, ParallelConfig
-        with joblib.parallel_config(verbose=100):
-            config = ParallelConfig(backend="joblib")
-            parallel_backend = init_parallel_backend(config)
+        from pydvl.parallel import JoblibParallelBackend
+        parallel_backend = JoblibParallelBackend()
         ```
-
-    Args:
-        config: instance of [ParallelConfig][pydvl.utils.config.ParallelConfig]
-            with cluster address, number of cpus, etc.
     """
 
-    def __init__(self, config: ParallelConfig):
+    @deprecated(
+        target=True,
+        args_mapping={"config": None},
+        deprecated_in="0.9.0",
+        remove_in="0.10.0",
+    )
+    def __init__(self, config: Optional[ParallelConfig] = None) -> None:
+        n_jobs: Optional[int] = None
+        if config is not None:
+            n_jobs = config.n_cpus_local
         self.config = {
-            "n_jobs": config.n_cpus_local,
+            "n_jobs": n_jobs,
         }
 
     @classmethod
     def executor(
         cls,
-        max_workers: int | None = None,
-        config: ParallelConfig = ParallelConfig(),
-        cancel_futures: CancellationPolicy = CancellationPolicy.NONE,
+        max_workers: Optional[int] = None,
+        *,
+        config: Optional[ParallelConfig] = None,
+        cancel_futures: Union[CancellationPolicy, bool] = CancellationPolicy.NONE,
     ) -> Executor:
+        """Returns a futures executor for the parallel backend.
+
+        !!! Example
+            ``` python
+            from pydvl.parallel import JoblibParallelBackend
+            parallel_backend = JoblibParallelBackend()
+            with parallel_backend.executor() as executor:
+                executor.submit(...)
```

Comment:
Your suggestion is exactly what we had before this PR with `init_parallel_backend`. I think the registry pattern hurts discoverability and readability and it's better to be more explicit in this case especially since changing this is considered "advanced usage" and the user should know what they're doing.

Reply:
No it's not "exactly what we had before"! I don't suggest going back, just a convenience function in case I want a factory to use based e.g. on whatever comes out of `os.getenv()`. But if you hate it, I guess people can just write an if/else

Reply:
I changed the deprecation for `init_parallel_backend` to only the config argument and allowed passing the backend name as a string.

Reply:
I have added a small snippet to the documentation about it.